sqldelight-idea-plugin/src/main/kotlin/com/squareup/sqldelight/intellij/lang/SqlDelightStructureViewFactory.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sqldelight.intellij.lang
+
+import com.alecstrong.sqlite.psi.core.psi.SqliteTypes
+import com.intellij.ide.structureView.StructureViewModel
+import com.intellij.ide.structureView.StructureViewModelBase
+import com.intellij.ide.structureView.StructureViewTreeElement
+import com.intellij.ide.structureView.TreeBasedStructureViewBuilder
+import com.intellij.ide.util.treeView.smartTree.TreeElement
+import com.intellij.lang.PsiStructureViewFactory
+import com.intellij.navigation.ItemPresentation
+import com.intellij.navigation.NavigationItem
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.util.Iconable
+import com.intellij.pom.Navigatable
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.squareup.sqldelight.core.SqlDelightTypes
+import com.squareup.sqldelight.core.lang.SqlDelightFile
+import com.squareup.sqldelight.core.lang.util.childrenOfType
+import javax.swing.Icon
+
+class SqlDelightStructureViewFactory : PsiStructureViewFactory {
+  override fun getStructureViewBuilder(psiFile: PsiFile) = object : TreeBasedStructureViewBuilder() {
+    override fun createStructureViewModel(editor: Editor?) = SqlDelightStructureViewModel(psiFile)
+    override fun isRootNodeShown() = false
+  }
+}
+
+internal class SqlDelightStructureViewModel(
+  psiFile: PsiFile
+) : StructureViewModelBase(psiFile, SqlDelightStructureViewElement(psiFile)),
+    StructureViewModel.ElementInfoProvider {
+  override fun isAlwaysLeaf(element: StructureViewTreeElement) = false
+  override fun isAlwaysShowsPlus(element: StructureViewTreeElement) = false
+}
+
+internal class SqlDelightStructureViewElement(
+  private val element: PsiElement
+) : StructureViewTreeElement,
+    ItemPresentation,
+    NavigationItem {
+  override fun getValue() = element
+  override fun getChildren() = when (element) {
+    is SqlDelightFile -> element.childIdentifiers()
+    else -> emptyArray()
+  }
+
+  override fun canNavigateToSource() = (element as Navigatable).canNavigateToSource()
+  override fun canNavigate() = (element as Navigatable).canNavigate()
+  override fun navigate(requestFocus: Boolean) = (element as Navigatable).navigate(requestFocus)
+  override fun getPresentation() = this
+  override fun getName(): String? = when {
```

Comment:
can be `when(element) { is SqlDelightFile... }` style with other changes

also you'll need `is SqliteViewName -> "CREATE VIEW ${element.name}"` and others once you add them to structure view